src/interceptors/logging.interceptor.ts

```diff
@@ -80,20 +80,88 @@ export class LoggingInterceptor implements NestInterceptor {
+            }
+        }
+
         if (outFields === true || outFields === '*' || outFields === 'all') {
-          return response;
+            return resp;
         }
 
-        return Object.keys(response).reduce((acc, key) => {
+        return Object.keys(resp).reduce((acc, key) => {
             if (outFields.includes(key)) {
-                acc[key] = response[key];
+                acc[key] = resp[key];
             }
             return acc;
         }, {});
     }
+
+    filterRequestParams(request: any, bodyOutFields: AuditLogMetadata['bodyOutFields'], paramsOutFields: AuditLogMetadata['paramsOutFields'], queryOutFields: AuditLogMetadata['queryOutFields'], requestInFields: AuditLogMetadata['requestInFields']) {
+        if (!paramsOutFields && !bodyOutFields && !queryOutFields) {
+            return {};
+        }
+
+        const requestParams = request.params;
+        const requestBody = request.body;
+        const requestQuery = request.query;
+        let result = {};
+
+        if (paramsOutFields === '*' || paramsOutFields === 'all') {
+            result = { ...requestParams }
+        }
+
+        if (bodyOutFields === '*' || bodyOutFields === 'all') {
+            result = { ...result, ...requestBody }
+        }
+
+        if (queryOutFields === '*' || queryOutFields === 'all') {
+            result = { ...result, ...requestQuery }
+        }
+
+        if (paramsOutFields !== '*' && paramsOutFields !== 'all' && paramsOutFields.length && Object.getPrototypeOf(requestParams) === Object.prototype) {
+            const data = Object.keys(requestParams).reduce((acc, key) => {
+                if (paramsOutFields.includes(key)) {
+                    acc[key] = requestParams[key];
+                }
+                return acc;
+            }, {});
+            result = { ...result, ...data }
+        }
+
+        if (bodyOutFields !== '*' && bodyOutFields !== 'all' && bodyOutFields.length && Object.getPrototypeOf(requestBody) === Object.prototype) {
+            const data = Object.keys(requestBody).reduce((acc, key) => {
+                if (bodyOutFields.includes(key)) {
+                    acc[key] = requestBody[key];
+                }
+                return acc;
+            }, {});
+            result = { ...result, ...data }
+        }
+
+        if (queryOutFields !== '*' && queryOutFields !== 'all' && queryOutFields.length && Object.getPrototypeOf(requestQuery) === Object.prototype) {
+            const data = Object.keys(requestQuery).reduce((acc, key) => {
+                if (queryOutFields.includes(key)) {
+                    acc[key] = requestQuery[key];
+                }
+                return acc;
+            }, {});
+            result = { ...result, ...data }
+        }
+        if (requestInFields && requestInFields.length) {
+            for (let i = 0; i < requestInFields.length; i++) {
+                delete result[requestInFields[i]]
+            }
+        }
```

Comment:
I think a helper function could trim this down a lot:
```suggestion
    const getFieldValues = (
      fields: string[] | '*' | 'all' = [],
      data: unknown,
    ) => {
      if (Object.getPrototypeOf(data) === Object.prototype) {
        if (fields === '*' || fields === 'all') {
          return data as Record<string, unknown>;
        }
        return Object.keys(data).reduce((acc, key) => {
          if (fields.includes(key) && !requestInFields?.includes(key)) {
            acc[key] = data[key];
          }
          return acc;
        }, {});
      }
      return {};
    };

    return {
      ...getFieldValues(paramsOutFields, requestParams),
      ...getFieldValues(bodyOutFields, requestBody),
      ...getFieldValues(queryOutFields, requestQuery),
    };
```